packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -704,6 +705,170 @@ describe('TransactionController', () => {
     });
   });
 
+  describe('with actionId', () => {
+    it('adds multiple transactions with same actionId and ensures second transaction result does not resolves before the first transaction result', async () => {
+      const controller = newController();
+
+      const mockOrigin = 'origin';
+      let firstTransactionCompleted = false;
+      let secondTransactionCompleted = false;
+
+      const { result: firstResult } = await controller.addTransaction(
+        {
+          from: ACCOUNT_MOCK,
+          to: ACCOUNT_MOCK,
+        },
+        {
+          origin: mockOrigin,
+          actionId: ACTION_ID_MOCK,
+        },
+      );
+
+      firstResult
+        .then(() => {
+          firstTransactionCompleted = true;
+        })
+        .catch(() => undefined);
+
+      const { result: secondResult } = await controller.addTransaction(
+        {
+          from: ACCOUNT_MOCK,
+          to: ACCOUNT_MOCK,
+        },
+        {
+          origin: mockOrigin,
+          actionId: ACTION_ID_MOCK,
+        },
+      );
+      secondResult
+        .then(() => {
+          secondTransactionCompleted = true;
+        })
+        .catch(() => undefined);
+
+      expect(firstTransactionCompleted).toBe(false);
```

Comment:
Just to be thorough, can we add a `flushPromises` before these assertions to ensure the event loop has had chance to run our `then` callbacks if the results had already resolved?